<launch>   
  <arg name="kinova_only"              default="false"/>
  <arg name="use_joy"                  default="false"/>

  <param name="urdf_path" type="string" value="$(find spot_kinova_description)" />
  <param name="urdf_name" type="string" value="/urdf/kinova_only.urdf" if="$(arg kinova_only)"/>
  <param name="urdf_name" type="string" value="/urdf/spot_kinova_simul.urdf" unless="$(arg kinova_only)"/>
  <param name="kinova_only" type="bool" value="true" if="$(arg kinova_only)"/>
  <param name="kinova_only" type="bool" value="false" unless="$(arg kinova_only)"/>
  <param name="use_joy" type="bool" value="true" if="$(arg use_joy)"/>
  <param name="use_joy" type="bool" value="false" unless="$(arg use_joy)"/>
  
  <node name="kimm_spot_kinova_test" pkg="kimm_spot_kinova" args="-keyboard" type="kimm_spot_kinova_test" output="screen">
  </node>

  <arg if="$(arg kinova_only)"  name="rviz_ref_frame" value="base_link"/>
  <arg unless="$(arg kinova_only)" name="rviz_ref_frame" value="odom"/>

  <group unless="$(arg kinova_only)">
    <arg name="robot_name"              default="/"/>  <!-- Change this for namespacing. -->
    <arg name="base_frame"             default="body"/> <!-- Link name of floating base. Do not touch this. -->
    <arg name="joints_map_file"        default="$(find kimm_spot_kinova)/../config/joints/joints.yaml"/> <!--Path to list of joint names. Do not touch this. -->
    <arg name="links_map_file"         default="$(find kimm_spot_kinova)/../config/links/links.yaml"/> <!-- Path to list of link names. Do not touch this. -->
    <arg name="gait_config_file"       default="$(find kimm_spot_kinova)/../config/gait/gait.yaml"/>  <!-- Path to gait parameters. Do not touch this. -->   
    <arg name="gazebo"                 default="false" /> <!-- Set to true during simulation. This is auto-set to true from gazebo.launch. -->
    <arg name="rviz"                   default="false"/> <!-- Set to true to run rviz in parallel. -->
    <arg name="has_imu"                default="true" /> <!-- Set to true if you want to visualize robot but there's no IMU. Only useful for microcontrollers. -->
    <arg name="lite"                   default="false" /> <!-- Set to true if you're using CHAMP lite version. Only useful for microcontrollers. -->
    <arg name="close_loop_odom"        default="true" /> <!-- Set to true if you want to calculate odometry using close loop. This is auto-set to true from gazebo.launch. -->
    <arg name="publish_foot_contacts"  default="true" /> <!-- Set to true if you want the controller to publish the foot contact states. This is auto-set to false from gazebo.launch. -->
    <arg name="publish_joint_control"  default="true" /> <!-- Set to true if you want the controller to publish the joint_states topic. This is auto-set to false from gazebo.launch. -->
    <arg name="laser"                  default="sim"/> <!-- Set to the 2D LIDAR you're using. See https://github.com/chvmp/champ/tree/master/champ_bringup/launch/include/laser .-->
    <arg name="joint_controller_topic" default="joint_group_position_controller/command" /> <!-- Change to remap command topic for actuator controller (ROS control).  -->
    <arg name="hardware_connected"     default="false" /> <!-- Flag useful to launch hardware connected launch files. This auto disables publishing joint_states. -->
    <arg name="description_file" default="$(find spot_kinova_description)/urdf/spot_kinova_simul.urdf"/>
 
    <arg if="$(eval arg('robot_name') == '/')"  name="frame_prefix" value="" />
    <arg unless="$(eval arg('robot_name') == '/')" name="frame_prefix" value="$(arg robot_name)/" />

    <include file="$(find champ_bringup)/launch/bringup.launch">
        <arg name="robot_name"             value="$(arg robot_name)"/>
        <arg name="base_frame"             value="$(arg base_frame)"/>
        <arg name="joints_map_file"        value="$(arg joints_map_file)"/>
        <arg name="links_map_file"         value="$(arg links_map_file)"/>
        <arg name="gait_config_file"       value="$(arg gait_config_file)"/>
        <arg name="description_file"       value="$(arg description_file)"/>
        <arg name="has_imu"                value="$(arg has_imu)"/>
        <arg name="gazebo"                 value="$(arg gazebo)"/>
        <arg name="lite"                   value="$(arg lite)"/>
        <arg name="laser"                  value="$(arg laser)"/>
        <arg name="rviz"                   value="$(arg rviz)"/>
        <arg name="rviz_ref_frame"         value="$(arg frame_prefix)$(arg rviz_ref_frame)"/>
        <arg name="joint_controller_topic" value="$(arg joint_controller_topic)" />
        <arg name="hardware_connected"     value="$(arg hardware_connected)" />
        <arg name="publish_foot_contacts"  value="$(arg publish_foot_contacts)" />
        <arg name="publish_joint_control"  value="$(arg publish_joint_control)" />
        <arg name="close_loop_odom"        value="$(arg close_loop_odom)" />
    
    </include>
  </group>

  <node name="rviz" pkg="rviz" type="rviz"
          args="-d $(find spot_viz)/rviz/robot.rviz -f $(arg rviz_ref_frame)"
    output="screen"/>
  <node name="arm_robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false" output="screen">
        <param name="use_tf_static"     value="false"/>
        <param name="publish_frequency" value="100"/>
        <param name="ignore_timestamp" value="true"/>
        <remap from="joint_states"     to="/arm/joint_states" />
  </node>

  <group if="$(arg use_joy)">
    <include file="$(find kimm_spot_teleop)/launch/teleop.launch">
      <arg name="joy"             value="true"/>
    </include>
  </group>

</launch>
